### ISSUE TITLE
Allow usage of failure::Error/std without backtrace on 0.1.X?

### ISSUE THREAD

**Issue by fitzgen** [withoutboats/failure#175]
AFAICT, this is not possible right now, but it is desirable to me when targeting wasm, which doesn't have backtrace support but can have alloc types and generally limp along with std. Alternatively, it would be nice if `Error` could be used without `std` and just the alloc feature/types.\

Happy to implement this with a little guidance. The dtolnay trick makes the code a bit hard to follow for me now.

**Comment by withoutboats** [comment]
Why do you need it on 0.1?

**Comment by fitzgen** [comment]
So I can use it now, rather than wait for 1.0

**Comment by withoutboats** [comment]
Ah, well there's no way on the released versions of failure on crates.io, and 1.0 will be the next released version (on March 15).

**Comment by fitzgen** [comment]
Hm. That's unfortunate, but I understand.
